components/hab/src/command/pkg/uninstall.rs

```diff
@@ -162,6 +181,15 @@ fn maybe_delete(
         return Ok(false);
```

Comment:
As an aside, we should probably add some logic so can't blow away the Launcher or Supervisor, either.

Perhaps in another PR, though 😄

Reply:
If it's not running, they get auto-downloaded on the next `hap sup run` so I was less worried about them as compare to `hab` itself.

Even if we delete it and you then do a restart of hab sup then things should just work but it's probably cleaner if we add them in to the services exclude list.  I'll need to work out how to get the running version of the sup which I don't think I can get right now.

Reply:
Accurately getting the version that's *running* would require a bit more work, but if you want to get the version of whatever `hab sup` executes, that shouldn't be too hard. You may want to crib some of the code for where [the launcher checks the version before launching](https://github.com/habitat-sh/habitat/blob/master/components/launcher/src/server/mod.rs#L534-L562).

Also, generally the version of the supervisor is the same as the `hab` binary.